systems/plants/RigidBodyManipulator.cpp

```diff
@@ -363,12 +363,12 @@ void RigidBodyManipulator::compile(void)
 {
   // reorder body list to make sure that parents before children in the list
 	size_t i=0;
-  while (i<bodies.size()) {
+  while (i<bodies.size()-1) {
```

Comment:
size_t is the right thing if you are comparing to the size of a vector, etc.  
the change in #1054 was because size_t was used for the body_index, which is a good old-fashioned int.